### PR TITLE
[posix] vendor support with lib dependencies

### PR DESCRIPTION
**Pull request by bertoldi-silabs**
A vendor interface implementation for the posix library is currently supported through linking a specified c++ source file. However, this is unable to accomodate dependencies for external libraries or header files that this vendor implementation source file may need.

The posix vendor support is changed to allow a vendor to specify a cmake RcpVendorDeps module file to resolve missing dependencies for a provided vendor support source file.

Additionally, this commit adds the ability for a vendor to define a radio url help message when the vendor rcp bus is selected instead of defaulting to the UART help message.

### CODE REVIEW

**Comment by bertoldi-silabs** [comment]
cc for code review: @jwhui @abtink

File: src/posix/platform/vendor.cmake
Comment (bukepo) [inline src/posix/platform/vendor.cmake:32]:
```suggestion
if(OT_POSIX_CONFIG_RCP_BUS STREQUAL "VENDOR")
```

File: src/posix/platform/vendor.cmake
Comment (bukepo) [inline src/posix/platform/vendor.cmake:49]: could you extend the existing check to cover the `RcpVendorDeps`?

File: src/posix/platform/vendor.cmake
Comment (zhanglongxia) [inline src/posix/platform/vendor.cmake:51]: nit: 4 spaces align

File: src/posix/platform/vendor.cmake
Comment (zhanglongxia) [inline src/posix/platform/vendor.cmake:30]: nit: 4 spaces align

File: src/posix/platform/vendor.cmake
Comment (bertoldi-silabs) [reply src/posix/platform/vendor.cmake:49]: I am not sure I understand specifically what the ask is for. Could you please clarify?

Is the idea something like this?
```
if(NOT TARGET RcpVendorDeps::RcpVendorDeps)
    find_package(RcpVendorDeps)
    ...
endif()
```

Comment (bukepo) [reply src/posix/platform/vendor.cmake:49]: I meant whether it's possible to make an example RcpVendorDeps to validate this works. Besides, Is it better make the name of the package a CMake variable like `OT_RCP_VENDOR_DEPS_PACKAGE`?

Comment (bertoldi-silabs) [reply src/posix/platform/vendor.cmake]: Thanks for the suggestion. I appreciate this approach for a vendor extension use case, as opposed to more standard fixed named packages.
Please let me know if the updated example fits what you were hoping to see.

**Comment by romacdon** [comment]
@bukepo - would it be possible to get another review on this PR?

CC: @abtink, @jwhui - perhaps you guys could review too.

File: src/posix/platform/FindExampleVendorDeps.cmake
Comment (bukepo) [inline src/posix/platform/FindExampleVendorDeps.cmake:57]:
```suggestion
This will define the following variables:
```

**Review by bukepo — APPROVED** [review]

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Comment by bertoldi-silabs** [comment]
@jwhui I believe that this PR is rebased up to date, and is ready to merge aside from the failing checks.
Could the failed checks please be rerun, and could the PR be merged pending once everything passes?

**Comment by bertoldi-silabs** [comment]
@jwhui This PR is up to date and ready to be merged. Could you please close and merge this PR when you have the chance?

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**
